api/WebApi.ts

```diff
@@ -42,7 +42,8 @@ import os = require('os');
 import url = require('url');
 import path = require('path');
 
-const isBrowser: boolean = typeof window !== 'undefined';
+const isNode = typeof process !== 'undefined' && process.release.name === 'node';
+const isBrowser: boolean = typeof window !== 'undefined' && !isNode;
```

Comment:
Isn't this going to be `false` in the renderer? I thought you mentioned something about that.

Reply:
Yes it will be, good point. I really don't know what we're supposed to do here.

Reply:
Like, it seems like something will always be wrong... Unless we just let the package consumer just tell us whether they're node or not.

Reply:
This is a little janky but I think it's the best I can come up with (kudos to [electron/electron#2288](https://github.com/electron/electron/issues/2288))
```js
const isElectron: boolean = process && process.versions && process.versions.hasOwnProperty('electron');
const isElectronWebpage: boolean = isElectron && process.type === 'renderer';
const isBrowser: boolean = typeof window !== 'undefined' && (!isElectron || isElectronWebpage);
```

Reply:
P.S. Electron can sandbox renderers heavily so we can't really trust a renderer to say "I am node!" (although we don't do this in GK and I doubt we ever would in anything running this). I don't know if workers can be sandboxed like that, but if so, workers would say they're neither node nor a browser.... womp womp. But I'm not going to worry too hard about that case, this is already an extreme edge case.